### PR TITLE
Feat/bonus

### PR DESCRIPTION
**Pull request by aurelien-boch**

### CODE REVIEW

**Review by Pepiloto — APPROVED** [review]

**Review by Pepiloto — APPROVED** [review]

File: api/src/api/recipe/rate.ts [hunk 1]
```diff
@@ -17,8 +17,10 @@ const rate = async (params: Params, user: User): Promise<EndpointReturn<never>>
         where: {id: params.recipeId },
         relations: ["owner"]
     });
+    console.log("before singleton")
     const pushNotification = pushNotificationProvider();
-
+   console.log(pushNotification);
+    console.log("before checks")
```
Comment (aurelien-boch) [inline api/src/api/recipe/rate.ts:23]: remove useless log

File: api/src/api/recipe/rate.ts [hunk 2]
```diff
@@ -17,8 +17,10 @@ const rate = async (params: Params, user: User): Promise<EndpointReturn<never>>
         where: {id: params.recipeId },
         relations: ["owner"]
     });
+    console.log("before singleton")
```
Comment (aurelien-boch) [inline api/src/api/recipe/rate.ts:20]: remove useless log

File: api/src/api/recipe/rate.ts [hunk 3]
```diff
@@ -31,21 +33,25 @@ const rate = async (params: Params, user: User): Promise<EndpointReturn<never>>
             errorMessage: "You cannot rate your own recipe"
         }
     }
+    console.log("before rating creation")
```
Comment (aurelien-boch) [inline api/src/api/recipe/rate.ts:36]: remove useless log

File: api/src/api/recipe/rate.ts [hunk 4]
```diff
@@ -31,21 +33,25 @@ const rate = async (params: Params, user: User): Promise<EndpointReturn<never>>
...
     const rating = ratingRepository.create({
         rating: params.rating,
         recipe,
         user
     });
 
+    console.log("before rating")
```
Comment (aurelien-boch) [inline api/src/api/recipe/rate.ts:43]: remove useless log

File: api/src/api/recipe/rate.ts [hunk 5]
```diff
@@ -31,21 +33,25 @@ const rate = async (params: Params, user: User): Promise<EndpointReturn<never>>
...
     await ratingRepository.save(rating);
+    console.log("after rating")
```
Comment (aurelien-boch) [inline api/src/api/recipe/rate.ts:45]: remove useless log

File: api/src/api/recipe/rate.ts [hunk 6]
```diff
@@ -31,21 +33,25 @@ const rate = async (params: Params, user: User): Promise<EndpointReturn<never>>
...
 
+    console.log(recipe.owner);
```
Comment (aurelien-boch) [inline api/src/api/recipe/rate.ts:47]: remove useless log

File: api/src/api/recipe/rate.ts [hunk 7]
```diff
@@ -31,21 +33,25 @@ const rate = async (params: Params, user: User): Promise<EndpointReturn<never>>
...
     pushNotification.sendNotification({
         message: `Your recipe ${recipe.name} has been rated by ${user.email} ! Click here to see the rating.`,
         userEndpoint: recipe.owner.notificationLink,
         url: '/recipe/' + recipe.id,
         title: 'One of your recipe has been rated!',
     })
-
+    console.log("after notif send");
```
Comment (aurelien-boch) [inline api/src/api/recipe/rate.ts:54]: remove useless log

File: frontend/src/pages/recipe/index.tsx [hunk 8]
```diff
@@ -88,7 +90,7 @@ const RatingDialog = ({currentRating, recipeId}: RatingDialogProps) => {
                     variant={"outlined"}
                 >Cancel</Button>
                 <Button
-                    onClick={() => mutate({recipeId, rating: rating as any})}
+                    onClick={() => mutate({ recipeId, rating: rating as any })}
```
Comment (aurelien-boch) [inline frontend/src/pages/recipe/index.tsx:93]: why cast

**Pull request merged**